### ISSUE TITLE
[Enhancement] publish on pub.dev please?

### ISSUE THREAD

**Issue by milindgoel15** [re7gog/android_system_font#1]
Great package, however the only way to add this is to clone this and add it locally. would be better via pub.dev

**Comment by timmyrosen** [comment]
@milindgoel15 Have you gotten this to work properly? Do you mind sharing your solution? I only get the `/system/fonts/Roboto-Regular.ttf` font path returned no matter what.

**Comment by milindgoel15** [comment]
this is my solution below. i actually made a stateful wrapper widget for this.

```dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:io';

import 'package:android_system_font/android_system_font.dart';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

class FontLoaderWidget extends StatefulWidget {
  const FontLoaderWidget({
    super.key,
    required this.child,
  });

  final Widget child;

  @override
  State<FontLoaderWidget> createState() => _FontLoaderWidgetState();
}

class _FontLoaderWidgetState extends State<FontLoaderWidget> {
  final _androidSystemFontPlugin = AndroidSystemFont();

  @override
  void initState() {
    super.initState();
    initPlatformState();
  }

  static Future<ByteData> _readFileBytes(String path) async {
    var bytes = await File(path).readAsBytes();
    return ByteData.view(bytes.buffer);
  }

  // Platform messages are asynchronous, so we initialize in an async method.
  Future<void> initPlatformState() async {
    String fontFilePath;
    // Platform messages may fail, so we use a try/catch PlatformException.
    // We also handle the message potentially returning null.
    try {
      fontFilePath = await _androidSystemFontPlugin.getFilePath() ??
          'Unknown font file path';
    } on PlatformException {
      fontFilePath = 'Failed to get font file path.';
    }

    // If the widget was removed from the tree while the asynchronous platform
    // message was in flight, we want to discard the reply rather than calling
    // setState to update our non-existent appearance.
    if (!mounted) return;

    setState(() {
      var fontLoader = FontLoader('SystemFont');
      fontLoader.addFont(_readFileBytes(fontFilePath));
      fontLoader.load();
    });
  }

  @override
  Widget build(BuildContext context) {
    return widget.child;
  }
}

```

**Comment by timmyrosen** [comment]
@milindgoel15 Thank you!

Does this work on all Android devices for you? Even non-rooted devices?

**Comment by milindgoel15** [comment]
> @milindgoel15 Thank you!
> 
> Does this work on all Android devices for you? Even non-rooted devices?

Yes

**Comment by timmyrosen** [comment]
@milindgoel15 That is so weird. I copied your widget and still only get `/system/fonts/Roboto-Regular.ttf` returned.

Do you mind sharing a bit more of your solution?
Where in the Widget tree do you use this FontLoaderWidget?
How do you specify the font in the theme?
Can you give an example of a device + OS version where this works?

**Comment by milindgoel15** [comment]
> Do you mind sharing a bit more of your solution?
> Where in the Widget tree do you use this FontLoaderWidget?

Basically keep the fontLoader as a parent to MaterialApp widget

            return FontLoaderWidget(
              child: androidMaterialApp(
                defaultTheme,
                context,
                dynamicMode,
                lightDynamic,
                primaryColorSwatch,
                isGoogleFont,
                isSystemFont,
                fontFamily,
                isAmoledMode,
                darkDynamic,
              ),
            );

> How do you specify the font in the theme?
So within the them configs (if you also have them separately)

    fontFamily: isGoogleFont == false
        ? isSystemFont
            ? "SystemFont"
            : "CircularStd"
        : "",

> Can you give an example of a device + OS version where this works?

I am using xiaomi + android 15. (Non-root, MIUI)

**Comment by timmyrosen** [comment]
@milindgoel15 That is basically exactly what I am doing. I haven't tried on a Xiaomi device though, but for me it doesn't work on Samsung, OnePlus or Nothing Phone. All running Android 14/15.

**Comment by milindgoel15** [comment]
> @milindgoel15 That is basically exactly what I am doing. I haven't tried on a Xiaomi device though, but for me it doesn't work on Samsung, OnePlus or Nothing Phone. All running Android 14/15.
> 
> 

Probably OEM related issues. Cuz many OEMs don't allow setting fonts system wide unless you root them.

**Comment by timmyrosen** [comment]
@milindgoel15 Yeah, that's what I'm thinking too. It's super weird though, this is an official accessibility requirement in the EU, and there is no way to actually achieve it.
